pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -241,7 +243,10 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({state})"
 
     def __getstate__(self) -> dict:
-        d = super().__getstate__()
+        with self.sync_context():
+            d = deepcopy(super().__getstate__())
+        if not self.is_tensor:
+            del d["value"]
```

Comment:
Can we maybe use the `state_dict` here? This has automatic syncing built-in.

Then it would be something like 

```python
if self.is_tensor:
    d = self.state_dict()
else:
    d = self.value.state_dict()
```

Reply:
This won't work as we need the full getstate. state_dict is only a portion of it.